mimic/test/test_swift.py

```diff
@@ -119,74 +313,617 @@ def test_get_container(self):
-                                  body=BODY)
-        self.assertEqual(self.successResultOf(object_response).code,
-                         201)
-        container_response = self.successResultOf(
-            request(self, self.root, b"GET", uri)
+        # put the object
+        self.put_object(headers={b"content-type": [b"text/plain"]})
+
+        # get the container listing
+        container_response, container_contents = self.get_container()
+
+        # Validate container meta-data
+        self.assertEqual(
+            container_response.headers.getRawHeaders(
+                b"X-Container-Object-Count")[0], b"1"
         )
-        self.assertEqual(container_response.code, 200)
-        container_contents = self.successResultOf(
-            treq.json_content(container_response)
+        self.assertEqual(
+            container_response.headers.getRawHeaders(
+                b"X-Container-Bytes-Used")[0],
+            "{0}".format(self.object_size).encode("ascii")
         )
+        # Validate container response
         self.assertEqual(len(container_contents), 1)
-        self.assertEqual(container_contents[0]['name'], "testobject")
+        self.assertEqual(container_contents[0]['name'],
+                         self.object_path.decode('utf-8'))
         self.assertEqual(container_contents[0]['content_type'], "text/plain")
-        self.assertEqual(container_contents[0]['bytes'], len(BODY))
-        object_response = self.successResultOf(
-            request(self, self.root, b"GET", object_uri)
+        self.assertEqual(container_contents[0]['bytes'], self.object_size)
+
+        # Get the object
+        object_response, object_body = self.get_object(
+            object_path=self.object_uri)
+        self.assertEquals(object_body, self.object_data)
+
+    def test_put_object_path_based_name(self):
+        """
+        PUT object - object name is a path instead of a simple string
+        """
+        # generate some paths with depth, swift's object name field is greedy
+        # and consumes everything, including slashes, after the container name
+        object_paths = [
+            self.object_path,
+        ]
+        object_path_prefix = b""
+        for i in range(10):
+            object_path_prefix = (
+                object_path_prefix + "{0}".format(i).encode('utf-8') + b"/")
+            new_object_path = object_path_prefix + object_paths[0]
+            object_paths.append(new_object_path)
+
+        container_size = "{0}".format(self.object_size).encode("utf-8")
+        for object_path in object_paths:
+
+            object_uri = self.uri + b"/" + object_path
+            self.put_object(
+                object_path=object_uri,
+                headers={b"content-type": [b"text/plain"]},
+            )
+
+            container_response, container_contents = self.get_container()
+
+            # Validate container meta-data
+            self.assertEqual(
```

Comment:
Another good testing idiom you may want to consider in the future is "[one assert per test](http://www.mfranc.com/unit-testing/good-unit-test-one-assert/)".  When you make this many asserts in a test, in a loop, it can be very difficult to tease out what's going wrong when they start failing.  It's also harder to write the tests first this way, because you have to drop the whole implementation at once rather than incrementally one property at a time.